pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -179,6 +180,11 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		// Get the current resource list from discovery.
 		newResources := GetDeletableResources(discoveryClient)
 
+		// Filter resources that have been uninstalled and reinstalled stopped since the last sync.
+		// The following sync will recognize them as new resources and they will be restarted.
+		// If we do not filter them out, the informer that was previously stopped will not get restarted.
+		gc.dependencyGraphBuilder.filterStoppedResources(newResources)
```

Comment:
could maybe just move this to be "old resources is a field on gc, and then when something gets stopped it gets removed from old resources", but eh, if this works it's prob fine as-is

Reply:
The problem here is that `startMonitors` (where we listen for the doneCh to fire to indicate that a resource's informer has stopped) lives on the graph builder and can't add to a field that is on the gc. 

So I think without rearranging the code a bunch, it's simplest to just populate a set on the graph_builder and then the gc Sync will just consume that. This is what I've tried to do here, but let me know if you think there is an easier way.